src/main/java/org/jetbrains/nativecerts/mac/SecurityFrameworkUtil.java

```diff
@@ -118,31 +133,80 @@ static boolean isSelfSignedCertificate(X509Certificate certificate) {
         return true;
     }
 
+    private static boolean validateCertificate(SecurityFramework.SecCertificateRef certificateRef) {
+        SecurityFramework.SecPolicyRef policy = null;
+        SecurityFramework.SecTrustRefByReference secTrustRefByReference = null;
+        CoreFoundation.CFArrayRef subjCerts = null;
+        try {
+            final Pointer[] values = {certificateRef.getPointer()};
+            subjCerts = CoreFoundationExt.INSTANCE.CFArrayCreate(
+                    null, values, new CoreFoundation.CFIndex(1), null
+            );
+
+            secTrustRefByReference = new SecurityFramework.SecTrustRefByReference();
+
+            policy = SecurityFramework.INSTANCE.SecPolicyCreateSSL(false, null);
+            SecurityFramework.OSStatus rc = SecurityFramework.INSTANCE.SecTrustCreateWithCertificates(
+                    subjCerts, policy, secTrustRefByReference);
+            if (!SecurityFramework.OSStatus.errSecSuccess.equals(rc)) {
+                /* should never happen */
+                return false;
+            }
+
+            return SecurityFramework.INSTANCE.SecTrustEvaluateWithError(secTrustRefByReference.getSecTrustRef(), null);
```

Comment:
Could you please add a WARN message that certificated is not trusted thus not imported

Reply:
Its valid for a certificate to be not trusted, its logged elsewhere:

```
LOGGER.fine("Certificate '" + certificateDescription + "' has no trust settings and failed to validate against trusted roots");
```